src/kernel/klib.h

```diff
@@ -0,0 +1,37 @@
+/* in-kernel loadable library interface */
+#include "../klib/klib.h"
+#define KLIB_MAX_NAME 16
+
+typedef struct klib_mapping {
+    struct rmnode n; /* virtual */
+    u64 phys;
+    u64 flags;
+} *klib_mapping;
+
+typedef struct klib {
+    char name[KLIB_MAX_NAME];
+    table syms;
+    rangemap mappings;
+    buffer elf;
+} *klib;
+
+typedef closure_type(klib_handler, void, klib, status);
+typedef int (*klib_init)(void *md, klib_add_sym add_sym);
```

Comment:
How about adding a "kern_get_sym" parameter to this function, which can then be used by a module to load kernel symbols?
Commit https://github.com/nanovms/nanos/commit/cacc023798ffb8a70e2df3ffa0a9fb5bbbfb1557 from my (work in progress) https://github.com/nanovms/nanos/tree/feature/telemetry branch has a possible implementation for that.